File: two-pointers/lint-609/lint-609.js
const twoSum5 = function (nums, target) {
  //注意js sort时最好写一个comparator函数
  nums.sort((a, b) => a - b);

  let left = 0;
  let right = nums.length - 1;

  let count = 0;
  while (left < right) {
    const value = nums[left] + nums[right];

    if (value > target) {
      right -= 1;
    } else {
      count += right - left;
      left += 1;
    }
  }

  return count;
};

File: two-pointers/lint-609/lint-609.py
# https://www.lintcode.com/problem/two-sum-less-than-or-equal-to-target/description?_from=ladder&&fromId=1
"""
排序后双zhizhen
"""


class Solution:
    """
    @param nums: an array of integer
    @param target: an integer
    @return: an integer
    """

    def twoSum5(self, nums, target):
        # write your code here
        left, right = 0, len(nums) - 1
        count = 0
        nums.sort()
        while left < right:
            value = nums[left] + nums[right]
            if value > target:
                right -= 1
            else:
                count += right - left
                left += 1
        return count
